api/openapi-spec.yaml

```diff
@@ -0,0 +1,352 @@
+            application/json:
+              schema:
+                $ref: "#/components/schemas/EiffelEvent"
+        400:
+          description: Wrong type given for parameter
+          content: {}
+        401:
+          description: Unauthorized
+          content: {}
+        403:
+          description: Forbidden
+          content: {}
+        404:
+          description: The requested event is not found
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  errorMsg:
+                    type: string
+                    example: "The requested event is not found"
+        500:
+          description: Internal server issue
+          content: {}
+  /search/{id}:
+    post:
+      tags:
+      - API
+      summary: To get upstream/downstream events for an event based on the searchParameters
+        passed
+      operationId: searchUsingPOST
+      parameters:
+      - name: id
+        in: path
+        description: "Id of the event."
+        required: true
+        schema:
+          type: string
+      - name: limit
+        in: query
+        description: >
+          Determines the maximum amount of events to be fetched.
+          Use `-1` for unlimited.
+        schema:
+          type: integer
+          format: int32
+      - name: levels
+        in: query
+        description: >
+          Determines the maximum amount of levels to search.
+          Use `-1` for unlimited
+        schema:
+          type: integer
+          format: int32
+      requestBody:
+        description: |
+          Select what types of links you want upstream/downstream search to follow.
+
+          Examples:
+
+            * Select `CAUSE` if you only want the search to follow `CAUSE` links
+              disregarding other links.
+            * Select `CONTEXT` and `ACTIVITY_EXECUTION` if you want to follow both `CONTEXT`
+              and `ACTIVITY_EXECUTION` links.
+
+          Link Types:
+
+            - CAUSE
+            - CONTEXT
+            - FLOW_CONTEXT
+            - ACTIVITY_EXECUTION
+            - PREVIOUS_ACTIVITY_EXECUTION
+            - PREVIOUS_VERSION
+            - COMPOSITION
+            - ENVIRONMENT
+            - ARTIFACT
+            - SUBJECT
+            - ELEMENT
+            - BASE
+            - CHANGE
+            - TEST_SUITE_EXECUTION
+            - TEST_CASE_EXECUTION
+            - IUT
+            - TERC
+            - MODIFIED_ANNOUNCEMENT
+            - SUB_CONFIDENCE_LEVEL
+            - REUSED_ARTIFACT
+            - VERIFICATION_BASIS
+            - ALL
+
+            **Example**
+
+            In the following example `dlt` stands for downlink and `ult` stands for uplink
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SearchParameters'
+        required: false
+      responses:
+        200:
+          description: OK
+          content:
```

Comment:
This response example should probably have an event with links both downstream and upstream

Reply:
Fixed in ....